Release preparation

junniest opened the pull request:
### Documentation enhancements:
* [`phylo/README.md`](diffhunk://#diff-85404cc12849de175af2b1b03bb86ee594fcad6969532bd0f88bde3e71b01c60L1-L7): Overhauled the README to provide comprehensive documentation, including descriptions of functionality, examples, crate features, roadmap, and contributor information. Added sections on contributing, licensing, and citation.

### Feature renaming:
* [`phylo/Cargo.toml`](diffhunk://#diff-daeced5f6b41a74f1cb9fe9daa782a803c095e77da8e8ba591f46b4802d19569R15-R23): Renamed the `use-precomputed-test-results` feature to `precomputed-test-results` for consistency. Updated feature descriptions and added the `rust-version` key with a minimum supported Rust version of 1.82.0.
* [`phylo/src/optimisers/topo_optimiser_tests.rs`](diffhunk://#diff-dd57a89522836606762bfe2f92c15ef6bdef131ea686b42e78e5e5abb060aa63L26-R26): Updated feature flags from `use-precomputed-test-results` to `precomputed-test-results` in test macros for alignment with the renamed feature. [[1]](diffhunk://#diff-dd57a89522836606762bfe2f92c15ef6bdef131ea686b42e78e5e5abb060aa63L26-R26) [[2]](diffhunk://#diff-dd57a89522836606762bfe2f92c15ef6bdef131ea686b42e78e5e5abb060aa63L37-R37)

### Code clarification:
* [`phylo/src/optimisers/topo_optimiser.rs`](diffhunk://#diff-9029e743ae8aaadceae02647051d1c29c73bfa942e513a7122bad5ae21ae0b93R56-R89): Added detailed comments to the `run` method in the `TopologyOptimiser` implementation, explaining its functionality, panics, and return values. Included an example usage of the method.

junniest committed:
Add doctest to topo_optimiser run()

junniest committed:
Rename precomputed feature to comply w/ guidelines

Rename `use-precomputed-test-results` feature to `precomputed-test-results` to comply
with Rust crate development guidelines.

junniest committed:
Add minimal Rust version to Cargo.toml

junniest committed:
Initial version of the README

junniest committed:
Fix README

Fix README to describe Dima's future feature properly, add ARPIP paper link.

junniest committed:
Move README to base folder

junniest committed:
Add todo for unnecessary RefCell removal

junniest committed:
topo+optimiser doctest fix

MattesMrzik commented on README.md:
```diff
@@ -0,0 +1,185 @@
+    assert_eq!(unopt_cost, result.initial_cost);
+    assert!(result.final_cost > result.initial_cost);
+    assert!(result.iterations <= 100);
+    assert_eq!(result.cost.tree().len(), 9); // The initial tree has 9 nodes, 5 leaves and 4 internal nodes.
```
```suggestion
    assert_eq!(result.cost.tree().len(), 9); // The initial tree has 9 nodes, 5 leaves and 4 internal nodes, and so should the resulting tree.
```

MattesMrzik commented on README.md:
```diff
@@ -0,0 +1,185 @@
+    let optimiser = TopologyOptimiser::new(c);
+    let result = optimiser.run()?;
+    assert_eq!(unopt_cost, result.initial_cost);
+    assert!(result.final_cost > result.initial_cost);
```
Don't we usually want to minimize the cost function?

junniest replied:
Hm. At least for likelihoods this is not the case, we want to maximise the probability/likelihood, and so also maximise the log likelihood. For parsimony scores you are correct, we want to minimise the score, but that's why when I define the cost for parsimony scores I return the negative of it. But I think it should be documented in the cost function definitions, I will add a comment.

MattesMrzik commented on README.md:
```diff
@@ -0,0 +1,185 @@
+- `par-regraft`: Enable parallel regrafting operations using Rayon;
+- `par-regraft-chunk`: Enable chunked parallel regrafting;
+- `par-regraft-manual`: Enable manual parallel regrafting control;
+- `deterministic`: Ensure deterministic behavior for reproducible results (required for running tests);
```
```suggestion
- `deterministic`: Ensure deterministic behaviour for reproducible results (required for running tests);
```

To consistently use British English

junniest committed:
Fixes to messages to fix [lint](https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args)

MattesMrzik commented on README.md:
```diff
@@ -0,0 +1,185 @@
+
+<!-- [![Crates.io](https://img.shields.io/crates/v/phylo.svg)](https://crates.io/crates/phylo)
+[![Documentation](https://docs.rs/phylo/badge.svg)](https://docs.rs/phylo) -->
+[![License](https://img.shields.io/badge/license-MIT%20OR%20Apache--2.0-blue.svg)](https://github.com/acg-team/rust-phylo#license)
```
One could also add the passing tests, and %coverage badge. Or do that in a future release.

junniest replied:
Great idea!

MattesMrzik reviewed (commented):
looking good

junniest committed:
Fix spelling README.md

Co-authored-by: Mattes Mrzik <mattes@mrzik.de>

junniest committed:
Fix assertion comment in README example

Co-authored-by: Mattes Mrzik <mattes@mrzik.de>

junniest committed:
Add comment on cost function definition

Added comment to the cost function definition so that users know that the function is maximised
and should be negated if it should be minimised.

junniest committed:
Merge branch 'release-preparation' of github.com:acg-team/rust-phylo into release-preparation

junniest committed:
Fix spelling License -> Licence for BE

junniest committed:
Added badges for code coverage and passing tests
